SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -0,0 +1,226 @@
+#else
+	typedef UProperty UnrealProperty;
+	typedef UStrProperty UnrealStrProperty;
+	typedef UObjectProperty UnrealObjectProperty;
+#endif
+	for (TFieldIterator<UnrealProperty> It(Struct); It; ++It)
+	{
+		UnrealProperty* Property = *It;
+
+		FString VariableName = Property->GetName();
+		const void* Value = Property->ContainerPtrToValuePtr<uint8>(&EventMessage);
+
+		check(Property->ArrayDim == 1); // Arrays not handled yet
+
+		// convert the property to a FJsonValue
+		if (UnrealStrProperty *StringProperty = CastField<UnrealStrProperty>(Property))
+		{
+			AddTraceEventStringField(VariableName, StringProperty->GetPropertyValue(Value));
+		}
+		else if (UnrealObjectProperty* ObjectProperty = CastField<UnrealObjectProperty>(Property))
+		{
+			UObject* Object = ObjectProperty->GetPropertyValue(Value);
+			if (Object)
+			{
+				AddTraceEventStringField(VariableName, Object->GetName());
+				if (AActor* Actor = Cast<AActor>(Object))
+				{
+					FString KeyString = VariableName + TEXT("Position");
+					FString ValueString = Actor->GetTransform().GetTranslation().ToString();
+					AddTraceEventStringField(KeyString, ValueString);
+				}
+			}
+			else
+			{
+				AddTraceEventStringField(VariableName, "Null");
+			}
+		}
+		else // Default
+		{
+			FString StringValue;
+			Property->ExportTextItem(StringValue, Value, NULL, NULL, PPF_None);
+			AddTraceEventStringField(VariableName, StringValue);
+		}
+
+		// else if (UEnumProperty* EnumProperty = Cast<UEnumProperty>(Property))
+		// else if (UNumericProperty *NumericProperty = Cast<UNumericProperty>(Property))
+		// else if (UBoolProperty *BoolProperty = Cast<UBoolProperty>(Property))
+		// else if (UTextProperty *TextProperty = Cast<UTextProperty>(Property))
+		// else if (UArrayProperty *ArrayProperty = Cast<UArrayProperty>(Property))
+		// else if (USetProperty* SetProperty = Cast<USetProperty>(Property))
+		// else if (UMapProperty* MapProperty = Cast<UMapProperty>(Property))
+		// else if (UStructProperty *StructProperty = Cast<UStructProperty>(Property))
+
+	}
+
+	// TODO(EventTracer): implement and call AddTargetObjectInfoToEventData
+	// TODO(EventTracer): implement and call AddComponentIdInfoToEventData
+
+	TraceEvent.data = EventData;
+	Trace_EventTracer_AddEvent(EventTracer, &TraceEvent);
+	Trace_EventData_Destroy(EventData);
+
+	return CurrentSpanId;
+}
+
+void SpatialEventTracer::Enable(const FString& FileName)
+{
+	Trace_EventTracer_Parameters parameters = {};
+	parameters.user_data = this;
+	parameters.callback = &SpatialEventTracer::TraceCallback;
+	EventTracer = Trace_EventTracer_Create(&parameters);
+	Trace_EventTracer_Enable(EventTracer);
+
+	bEnabled = true;
+
+	UE_LOG(LogSpatialEventTracer, Log, TEXT("Spatial event tracing enabled."));
+
+	// Open a local file
+	const FString FolderPath = FPaths::Combine(FPaths::ProjectSavedDir(), TEXT("EventTracing"));
+	const FString FullFileName = FString::Printf(TEXT("EventTrace_%s_%s.trace"), *FileName, *FDateTime::Now().ToString());
```

Comment:
Hmm we should try and make the naming of these files consistent with the runtime ones. I think the runtime one is just called `event-tracing.log` at the moment. I'll have a think about a potentially better suffix than `.log` though because I think it makes it sound like a plain text log file.

- [ ] Decide upon event log file naming conventions

Reply:
Sure. I think the plan was to make the logs unique so they would not overwrite. Additionally adding in the workerId seemed a useful touch. I will let you decide.